File: backend/CookBook.Models/Ingredient.cs
namespace CookBook.Models {
    public enum IngredientType {
        Other,
        Vegetable,
        Fruit,
        Dairy,
        Meat,
        Fish,
        Grains,
        Breads,
        Nuts,
        Condiments,
        Herbs,
        Oils,
        Sweets,
    }

    public class Ingredient {
        public string Name { get; set; } = "";
        public string Description { get; set; } = "";
        public IngredientType Type { get; set; }
        public Quantity Quantity { get; set; } = new Quantity();
    }
}

File: frontend/cook-book/src/models/Ingredient.ts
import { Quantity } from "./Quantity";

export enum IngredientType {
    Other,
    Vegetable,
    Fruit,
    Dairy,
    Meat,
    Fish,
    Grains,
    Breads,
    Nuts,
    Condiments,
    Herbs,
    Oils,
    Sweets,
};

export interface Ingredient {
    name: string,
    description: string,
    type: IngredientType,
    quantity: Quantity
}